test_scripts/Polices/External_UCS_Informing_HMI/017_ATF_N_GetListofPermissions_empty_due_to_missing_entityID_OnAppPermissionChange.lua

```diff
@@ -0,0 +1,125 @@
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+local testCasesForPolicyTable = require('user_modules/shared_testcases/testCasesForPolicyTable')
+
+--[[ General Precondition before ATF start ]]
+commonFunctions:SDLForceStop()
+commonSteps:DeleteLogsFiles()
+commonSteps:DeletePolicyTable()
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('cardinalities')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_trigger_getting_device_consent()
+  testCasesForPolicyTable:trigger_getting_device_consent(self, config.application1.registerAppInterfaceParams.appName, config.deviceMAC)
+end
+
+function Test:Precondition_PTU_and_OnAppPermissionConsent_AllParams_Valid()
+  local ptu_file_path = "files/jsons/Policies/Related_HMI_API/"
+  local ptu_file = "OnAppPermissionConsent_ptu.json"
+  
+  testCasesForPolicyTable:flow_SUCCEESS_EXTERNAL_PROPRIETARY(self, nil, nil, nil, ptu_file_path, nil, ptu_file)
+  
+  EXPECT_HMINOTIFICATION("SDL.OnAppPermissionChanged",{ appID = self.applications[config.application1.registerAppInterfaceParams.appName]})
+  :Do(function(_,data)
+    if (data.params.appPermissionsConsentNeeded== true) then
+      local RequestIdListOfPermissions = self.hmiConnection:SendRequest("SDL.GetListOfPermissions", {appID = self.applications[config.application1.registerAppInterfaceParams.appName]})
+        EXPECT_HMIRESPONSE(RequestIdListOfPermissions,{result = {code = 0, method = "SDL.GetListOfPermissions",
+          -- allowed: If ommited - no information about User Consent is yet found for app.
+          allowedFunctions = {
+            { name = "Location", id = 156072572},
+            { name = "Notifications", id = 1809526495}
+          },
+          externalConsentStatus = {}
+        }
+      })
+      :Do(function()
+        local ReqIDGetUserFriendlyMessage = self.hmiConnection:SendRequest("SDL.GetUserFriendlyMessage",
+        {language = "EN-US", messageCodes = {"AppPermissions"}})
+        
+        EXPECT_HMIRESPONSE(ReqIDGetUserFriendlyMessage,
+        {result = {code = 0, messages = {{messageCode = "AppPermissions"}}, method = "SDL.GetUserFriendlyMessage"}})
+        :Do(function(_,_)
+          self.hmiConnection:SendNotification("SDL.OnAppPermissionConsent",
+          {
+            appID = self.applications[config.application1.registerAppInterfaceParams.appName],
+            consentedFunctions = {
+              { allowed = true, id = 156072572, name = "Location-1"},
+              { allowed = true, id = 1809526495, name = "Notifications"}
+            },
+            externalConsentStatus = {
+              {entityType = 13, status = "ON"}
+            },
+            source = "GUI"
+          })
+          EXPECT_NOTIFICATION("OnPermissionsChange")
```

Comment:
@vvvakulenko, "OnPermissionsChange" should not be sent because OnAppPermissionConsent is invalidated.See "[OnPermissionsChange] send after app's permissions change by Policy Table Update "
You can use:
EXPECT_NOTIFICATION("OnPermissionsChange"):Times(0)
commonTestCases:DelayedExp(10000)
DelayExp is added because ATF doesn't wait default timeout to elapse when Times is 0.